crates/bevy_render/src/color/mod.rs

```diff
@@ -406,6 +408,237 @@ impl Color {
         self
     }
 
+    // Linear RGB
+
+    /// Get the red in linear rgb colorspace.
+    pub fn linear_rgb_r(&self) -> f32 {
+        match self.as_rgba_linear() {
+            Color::RgbaLinear { red, .. } => red,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Get the green in linear rgb colorspace.
+    pub fn linear_rgb_g(&self) -> f32 {
+        match self.as_rgba_linear() {
+            Color::RgbaLinear { green, .. } => green,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Get the blue in linear rgb colorspace.
+    pub fn linear_rgb_b(&self) -> f32 {
+        match self.as_rgba_linear() {
+            Color::RgbaLinear { blue, .. } => blue,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Set the red in linear rgb colorspace.
+    pub fn set_linear_rgb_r(&mut self, r: f32) -> &mut Self {
+        *self = self.as_rgba_linear();
+        match self {
+            Color::RgbaLinear { red, .. } => *red = r,
+            _ => unreachable!(),
+        }
+        self
+    }
+
+    /// Returns this color with red set to a new value in linear rgb colorspace.
+    #[must_use]
+    pub fn with_linear_rgb_r(mut self, r: f32) -> Self {
+        self.set_linear_rgb_r(r);
+        self
+    }
+
+    /// Set the green in linear rgb colorspace.
+    pub fn set_linear_rgb_g(&mut self, g: f32) -> &mut Self {
+        *self = self.as_rgba_linear();
```

Comment:
Hmm... This changes the current color space of the color when writing to a channel in a different color space, which I found a bit surprising.

I personally think it would be more intuitive if the color was converted, the channel replaced then the color converted back to the original color space...

But can I see how that could be problematic when changing color channels of spaces that are surjective with regards the current color space (e.g. changing an HSL channel of an RGB color)

**Edit:** Looks like the current `set_*()` methods already behave like that? So it's not unprecedented. Maybe we could document this behavior?

Reply:
Added the following to the end of each doc comment for the setters.
```
returned value is of variant [`Color::XXX`]
```